protocols/migration/migration_reports_4_0_0_to_reports_5_0_0.py

```diff
@@ -0,0 +1,573 @@
+
+        # converts all reported variants
+        new_instance.variants = self.migrate_reported_variants(old_instance.candidateVariants, assembly)
+
+        # converts all analysis panels
+        if old_instance.additionalAnalysisPanels is not None:
+            panels = []
+            for panel in old_instance.additionalAnalysisPanels:
+                new_panel = self.new_model.AdditionalAnalysisPanel()  # :type reports_5_0_0.AdditionalAnalysisPanel
+                new_panel.specificDisease = panel.specificDisease
+                new_panel.panel = self.new_model.GenePanel(panelName=panel.panelName, panelVersion=panel.panelVersion)
+                panels.append(new_panel)
+            new_instance.additionalAnalysisPanels = panels
+
+        return self.validate_object(
+            object_to_validate=new_instance, object_type=self.new_model.ClinicalReportRD
+        )
+
+    def migrate_cancer_interpretation_request(self, old_instance):
+        """
+        :type old_instance: reports_4_0_0.CancerInterpretationRequest
+        :rtype: reports_5_0_0.CancerInterpretationRequest
+        """
+        raise NotImplemented
+
+    def migrate_cancer_interpretation_request_to_cancer_interpreted_genome(
+            self, old_instance, assembly, interpretation_service,
+            reference_database_versions, software_versions, report_url=None, comments=None):
+        """
+        NOTE: we migrate from a model where only one sample and one participant is supported, thus we do not need
+        a list of samples or participants
+        :type old_instance: reports_4_0_0.CancerInterpretationRequest
+        :type assembly: reports_5_0_0.Assembly
+        :type interpretation_service: str
+        :type reference_database_versions: dict
+        :type software_versions: dict
+        :type report_url: str
+        :type comments: list
+        :rtype: reports_5_0_0.CancerInterpretedGenome
+        """
+        new_instance = self.convert_class(
+            self.new_model.CancerInterpretedGenome, old_instance)  # :type: reports_5_0_0.CancerInterpretedGenome
+
+        # reportRequest id and version are interpretationRequest id and version
+        new_instance.interpretationRequestId = old_instance.reportRequestId
+        new_instance.interpretationRequestVersion = old_instance.reportVersion
+
+        # missing fields not existing in reports_4_0_0.InterpretationRequestRD will be received as parameters
+        new_instance.interpretationService = interpretation_service
+        new_instance.referenceDatabasesVersions = reference_database_versions
+        new_instance.softwareVersions = software_versions
+        new_instance.reportUrl = report_url
+        new_instance.comments = comments
+
+        # converts all reported variants
+        participant_id = old_instance.cancerParticipant.individualId
+        tumor_samples = old_instance.cancerParticipant.tumourSamples
+        if tumor_samples is None or len(tumor_samples) < 1:
```

Comment:
this could be reduced to:
```
if not tumor_samples:
```
which returns `True` if `tumour_samples` is `None` or an empty list `[]`